### PR TITLE
bugfix: return on error

### PR DESCRIPTION
**Pull request by cofyc**

### CODE REVIEW

**Comment by cofyc** [comment]
The error is not returned, users couldn't know why `additionalScrapeConfigs` is not appended in final prometheus.yaml.

cc @brancz

**Comment by mxinden** [comment]
I will go ahead and merge once CI runs green.

**Comment by brancz** [comment]
lgtm thanks a lot for the fix @cofyc !

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func generateConfig(p *v1.Prometheus, mons map[string]*v1.ServiceMonitor, basicA
 	var additionalScrapeConfigsYaml []yaml.MapSlice
 	err = yaml.Unmarshal([]byte(additionalScrapeConfigs), &additionalScrapeConfigsYaml)
 	if err != nil {
-		errors.Wrap(err, "unmarshalling additional scrape configs failed")
+		return errors.Wrap(err, "unmarshalling additional scrape configs failed")
```
Comment (mxinden) [inline pkg/prometheus/promcfg.go]: I forgot to check the function signature. You would need to do:

```go
return nil, errors.Wrap(err, "unmarshalling additional scrape configs failed")
```

(See [above](https://github.com/coreos/prometheus-operator/pull/1508/files#diff-d7fb441eb132cd7782e17b92b0eb5bcdR97))

File: pkg/prometheus/promcfg.go [hunk 2]
```diff
@@ -160,7 +160,7 @@ func generateConfig(p *v1.Prometheus, mons map[string]*v1.ServiceMonitor, basicA
 	var additionalAlertManagerConfigsYaml []yaml.MapSlice
 	err = yaml.Unmarshal([]byte(additionalAlertManagerConfigs), &additionalAlertManagerConfigsYaml)
 	if err != nil {
-		errors.Wrap(err, "unmarshalling additional alert manager configs failed")
+		return errors.Wrap(err, "unmarshalling additional alert manager configs failed")
```
Comment (mxinden) [inline pkg/prometheus/promcfg.go]: I forgot to check the function signature. You would need to do:

```go
return nil, errors.Wrap(err, "unmarshalling additional scrape configs failed")
```

(See [above](https://github.com/coreos/prometheus-operator/pull/1508/files#diff-d7fb441eb132cd7782e17b92b0eb5bcdR97))

**Comment by cofyc** [comment]
@mxinden Sorry, my bad, didn't test it locally.

I'm not sure why `TestStatefulSetVolumeInitial` test does not pass (skipped in CI?). I found `prometheusRuleFilesConfigMapName` function has been updated [here](https://github.com/coreos/prometheus-operator/commit/afe9711ce2ef5dd0e0b5091f921ae9bc0aab3711#diff-22c69a436e9da7a65888996fc2ab6d93R210), so I updated expected data in TestStatefulSetVolumeInitial.

**Comment by brancz** [comment]
huh strange, on master everything seems to be passing...

Anyways thanks a ton! This lgtm :+1:

**Pull request merged**